projects/core/src/site-context/site-context.module.ts

```diff
@@ -38,6 +40,18 @@ export function initSiteContextConfig(
   }
   return null;
 }
+export function CurrencyStatePersistenceFactory(
+  currencyPersistenceService: CurrencyStatePersistenceService
+) {
+  const result = () => currencyPersistenceService.initSync();
+  return result;
+}
+export function LanguageStatePersistenceFactory(
+  languagePersistenceService: LanguageStatePersistenceService
+) {
+  const result = () => languagePersistenceService.initSync();
+  return result;
+}
```

Comment:
We should not start syncing before we get the `context` config from CMS (aka. auto-config).
So please first await for promise `configInit.getStableConfig('context');` and only then call initSync().